NuRadioReco/modules/RNO_G/crRNOGTemplateCreator.py

```diff
@@ -0,0 +1,248 @@
+
+                Parameters
+                ----------
+                detector_file: string
+                    path to the detector file used for the template set
+                template_save_path: string
+                    path to a folder where the templates are stored
+                logger_level: string or logging variable
+                    Set verbosity level for logger (default: logging.NOTSET)
+                """
+
+        self.__detector_file = detector_file
+
+        self.logger.setLevel(logger_level)
+
+        # set up the efield to voltage converter
+        self.__efieldToVoltageConverter.begin(debug=False)
+
+        self.__debug = debug
+
+        self.__template_save_path = template_save_path
+
+    def set_template_parameter(self, template_run_id=[0, 0, 0], template_event_id=[0, 1, 2], template_station_id=[101, 101, 101], template_channel_id=[0, 0, 0], Efield_width=[5, 4, 2],
+                                antenna_rotation=[160, 160, 160], Efield_amplitudes=[-0.2, 0.8], cr_zenith=[55, 55, 55], cr_azimuth=[0, 0, 0],
+                                sampling_rate=3.2 * units.GHz, number_of_samples=2048):
+        """
+        set_parameter_templates method
+
+        sets the parameter to create the template set
+
+        Parameters
+        ----------
+        template_run_id: list of int
+            run ids of the artificial templates
+        template_event_id: list of int
+            event ids of the artificial templates
+        template_station_id:
+            station ids of the artificial templates
+        template_channel_id:
+            channel ids of the artificial templates
+        Efield_width: list of int
+            width (in samples) of the gaussian function used to create the Efield
+        antenna_rotation: list of int
+            rotation angle of the LPDA
+        Efield_ratio: list
+            array with the amplitudes of the Efield components [E_theta, E_phi]
+        cr_zenith: list of int
+            zenith angle of the cr for the template
+        cr_azimuth: list of int
+            azimuth angle of the cr for the template
+        sampling_rate: float
+            sampling rate used to build the template
+        number_of_samples: int
+            number of samples used for the trace
+        """
+
+        self.__template_run_id = template_run_id
+        self.__template_event_id = template_event_id
+        self.__template_station_id = template_station_id
+        self.__template_channel_id = template_channel_id
+
+        self.__Efield_width = Efield_width
+        self.__Efield_amplitudes = Efield_amplitudes
+        self.__antenna_rotation = antenna_rotation
+
+        self.__sampling_rate = sampling_rate
+        self.__template_sample_number = number_of_samples
+
+        self.__cr_zenith = cr_zenith
+        self.__cr_azimuth = cr_azimuth
+
+    def run(self, template_filename='templates_cr_station_101.p', include_hardware_response=True, return_templates=False):
```

Comment:
Are you sure that .p is a useful extension?

Reply:
I changed it to .pickle